activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1131,6 +1131,17 @@ def add_foreign_key(from_table, to_table, **options)
         return if options[:if_not_exists] == true && foreign_key_exists?(from_table, to_table, **options.slice(:column))
 
         options = foreign_key_options(from_table, to_table, options)
+
+        if options[:deferrable] == true
```

Comment:
I should have commented this in advance, I think this requirement is PostgreSQL adapter specific.
So how about creating `add_foreign_key(from_table, to_table, **options)` method in activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb to run these warnings validations?